test_all_sd_vm_apt_sources (test_vms_platform.SD_VM_Platform_Tests.test_all_sd_vm_apt_sources) CI failure

https://ws-ci-runner.securedrop.org/2024-04-11-235217913151-3ce1c0ad0f697cbaa0ff62b554c81eed2b04e079-Qubes_4.2_A-base_setup_20240410.log.txt

```
INFO:[2024-04-12-00:31:48:674953] ======================================================================
INFO:[2024-04-12-00:31:48:674984] FAIL: test_all_sd_vm_apt_sources (test_vms_platform.SD_VM_Platform_Tests.test_all_sd_vm_apt_sources)
INFO:[2024-04-12-00:31:48:675022] Test all VMs fpf apt source list iteratively.
INFO:[2024-04-12-00:31:48:675056] ----------------------------------------------------------------------
INFO:[2024-04-12-00:31:48:675096] Traceback (most recent call last):
INFO:[2024-04-12-00:31:48:675130]   File "/home/user/securedrop-workstation/tests/test_vms_platform.py", line 224, in test_all_sd_vm_apt_sources
INFO:[2024-04-12-00:31:48:675161]     self._validate_apt_sources(vm)
INFO:[2024-04-12-00:31:48:675196]   File "/home/user/securedrop-workstation/tests/test_vms_platform.py", line 83, in _validate_apt_sources
INFO:[2024-04-12-00:31:48:675230]     self.assertIn(self.apt_url, contents, f"{vm.name} missing apt sources list")
INFO:[2024-04-12-00:31:48:675261] AssertionError: 'deb [arch=amd64] https://apt-test.freedom.press bookworm nightlies' not found in 'deb [arch=amd64] https://apt-test.freedom.press bookworm main nightlies' : sd-gpg missing apt sources list
```

--- a/tests/test_vms_platform.py
+++ b/tests/test_vms_platform.py
@@ -23,18 +23,9 @@ class SD_VM_Platform_Tests(unittest.TestCase):
     def setUp(self):
         self.app = Qubes()
         with open("config.json") as c:
-            config = json.load(c)
-            if "environment" not in config:
-                config["environment"] = "dev"
-
-            dist = SUPPORTED_SD_DEBIAN_DIST
-
-            if config["environment"] == "prod":
-                self.apt_url = FPF_APT_SOURCES.format(dist=dist, component="main")
-            elif config["environment"] == "staging":
-                self.apt_url = FPF_APT_TEST_SOURCES.format(dist=dist, component="main")
-            else:
-                self.apt_url = FPF_APT_TEST_SOURCES.format(dist=dist, component="main nightlies")
+            self.config = json.load(c)
+        if "environment" not in self.config:
+            self.config["environment"] = "dev"
 
     def tearDown(self):
         pass
@@ -76,11 +67,20 @@ class SD_VM_Platform_Tests(unittest.TestCase):
         if vm.name in ["sd-whonix"]:
             return
 
+        if self.config["environment"] == "prod":
+            expected = FPF_APT_SOURCES.format(dist=SUPPORTED_SD_DEBIAN_DIST, component="main")
+        elif self.config["environment"] == "staging":
+            expected = FPF_APT_TEST_SOURCES.format(dist=SUPPORTED_SD_DEBIAN_DIST, component="main")
+        else:
+            expected = FPF_APT_TEST_SOURCES.format(
+                dist=SUPPORTED_SD_DEBIAN_DIST, component="main nightlies"
+            )
+
         cmd = "cat {}".format(APT_SOURCES_FILE)
         stdout, stderr = vm.run(cmd)
         contents = stdout.decode("utf-8").rstrip("\n")
 
-        self.assertIn(self.apt_url, contents, f"{vm.name} missing apt sources list")
+        self.assertEqual(expected, contents, f"{vm.name} missing apt sources list")
 
     def _ensure_packages_up_to_date(self, vm, fedora=False):
         """

--- a/tests/test_vms_platform.py
+++ b/tests/test_vms_platform.py
@@ -34,7 +34,7 @@ class SD_VM_Platform_Tests(unittest.TestCase):
             elif config["environment"] == "staging":
                 self.apt_url = FPF_APT_TEST_SOURCES.format(dist=dist, component="main")
             else:
-                self.apt_url = FPF_APT_TEST_SOURCES.format(dist=dist, component="nightlies")
+                self.apt_url = FPF_APT_TEST_SOURCES.format(dist=dist, component="main nightlies")
 
     def tearDown(self):
         pass

